pkg/api/injectsnippetsfs.go

```diff
@@ -0,0 +1,108 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"html"
+	"io"
+	"io/fs"
+	"time"
+)
+
+type InjectSnippetsFS struct {
+	fs.FS
+	name    string
+	content string
+}
+
+func NewInjectIndexFS(fsys fs.FS, name string, snippets map[string]string) (fs.FS, error) {
+	// no snippets, we serve using the name fs
+	if len(snippets) == 0 {
+		return fsys, nil
+	}
+
+	// read the content and inject snippets
+	f, err := fsys.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = f.Close() }()
+	all, err := io.ReadAll(f)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", name, err)
+	}
+
+	// inject snippets code into content
+	codeSnippets := renderCodeSnippets(snippets)
+	endElement := []byte("</head>")
+	idx := bytes.Index(all, endElement)
```

Comment:
I would match rather have an explicit marker for where to inject snippets.

Reply:
Sure, I had this version - but didn't want to show the marker or process the file in case there are no snippets